File: BDS/CSharp/Swizilla/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace Swizilla
{
    public struct Point
    {
        public int PointX { get; set; }
        public int PointY { get; set; }
        public Point(int pointX=0, int pointY=0)
        {
            PointX = pointX;
            PointY = pointY;
        }
    }

    public interface IShape
    {
        double calculateArea();
    }

    public class Rectangle : IShape
    {
        public Point StartPoint { get; set; }
        public Point EndPoint { get; set; }

        public Rectangle(Point startPoint =new Point(), Point endPoint=new Point())
        {
            StartPoint = startPoint;
            EndPoint = endPoint;
        }
        public double calculateArea()
        {
            return Math.Abs(EndPoint.PointX - StartPoint.PointX) * Math.Abs(EndPoint.PointY - StartPoint.PointY);
        }
    }

    public class Circle : IShape
    {
        public Point CenterPoint { get; set; }
        public double Radius;

        public Circle(Point centerPoint = new Point(), double radius=0)
        {
            CenterPoint = centerPoint;
            Radius = radius;
        }
        public double calculateArea()
        {
            return Math.PI * Math.Pow(Radius, 2);
        }
    }
    class ShapeCreator
    {
        public static IShape NewShape(string name)
        {
            switch (name.ToLower())
            {
                case "rectangle":
                case "rect":
                    return new Rectangle();

                case "circle":
                case "circ":
                    return new Circle();
            }
            return null;
        }
    }
}

File: BDS/Java/Swizilla/src/Shape.java
class Point {
	public int x;
	public int y;

	public int getX() {
		return x;
	}

	public void setX(int x) {
		this.x = x;
	}

	public int getY() {
		return y;
	}

	public void setY(int y) {
		this.y = y;
	}

	public Point(int x, int y) {
		this.x = x;
		this.y = y;
	}

	public Point() {
		this(0, 0);
	}

}

interface Shape {
	double calculateArea();
}

class Rectangle implements Shape {
	public Point startPoint;
	public Point endPoint;

	public Rectangle(Point startPoint, Point endPoint) {
		this.startPoint = startPoint;
		this.endPoint = endPoint;
	}
	
	public Rectangle() {
		this(new Point(), new Point());
	}

	public double calculateArea() {
		return Math.abs(endPoint.x - startPoint.x) * Math.abs(endPoint.y - startPoint.y);
	}
}

class Circle implements Shape
{
	public Point centerPoint;
	public double radius;


	public Point getCenterPoint() {
		return centerPoint;
	}

	public void setCenterPoint(Point centerPoint) {
		this.centerPoint = centerPoint;
	}

	public double getRadius() {
		return radius;
	}

	public void setRadius(double radius) {
		this.radius = radius;
	}


	public Circle(Point centerPoint, double radius) {
		this.centerPoint = centerPoint;
		this.radius = radius;
	}

	public Circle() {
		this(new Point(), 0.0);
	}
	
	public double calculateArea() {
		return Math.PI * Math.pow(radius, 2);
	}

}

class ShapeCreator {
	public static Shape newShape(String name) {
		switch (name.toLowerCase()) {
		case "rectangle":
		case "rect":
			return new Rectangle();

		case "circle":
		case "circ":
			return new Circle();
		}
		return null;
	}
}